src/sensai/evaluation/eval_stats/eval_stats_base.py

```diff
@@ -52,8 +57,7 @@ def __str__(self):
 
 
 class Metric(Generic[TEvalStats], ABC):
-    def __init__(self, name: str):
```

Comment:
Making the name static reduces generality as we lose the ability to have the name depend on the parametrisation of the metric.
Why did you change this?

Reply:
I explained it in the commit. My particular use case was that I needed the metric name for parsing and sorting but only had access to it when instantiating the object. Sure, the metric objects are lioghtweight, but it still felt strange having to instantiate them just for the name

Reply:
In any case, I knew that there would be a discussion about it, that's why the change is in one commit and easily revertible. When I changed it I hadn't realized the one exception where the name actually depended on the init parameters

Reply:
I guess I should read your commit messages (was in a hurry).
Anyway, I agree that it feels odd to instantiate a new object only to retrieve the name.
One solution would be for all metrics without arguments to be singletons. But that feels like a bit of overkill.
How about a hybrid solution like this (best of both worlds):

    class Foo:
        metricName: str

        def __init__(self, name=None):
            """
            :param name: the name of the metric; if None use the class' metricName attribute
            """
            self.name = name if name is not None else self.__class__.metricName


    class Bar(Foo):
        metricName = "BAR"


    class Baz(Foo):
        metricName = "BAZ"

        def __init__(self, param):
            super().__init__(name=f"BAZ{param}")


    class Wrong(Foo):
        pass  # will raise AttributeError

Unlike what you state, please note that your current solution is not fully backward compatible, since you dropped the constructor of Metric. A user who created his own Metric subclass and thus calls the super constructor gets an exception.

Reply:
Since I used __class__ to refer to the class attribute above, it could even also be called `name`.

Reply:
> I guess I should read your commit messages 

Is there even a way to view all changes and the corresponding commit messages at the same time (like IntelliJ's annotate)? I don't see it anywhere.
GitHub doesn't make it easy for reviewers...
One could review on a per-commit basis, but that's not the best idea in many cases.

Reply:
Yeah, this is really annoying. I don't know of a way for doing this in Github. That's why I mentioned this important commit in the PR's description - if you hover over the link with the mouse you'll see the message

Reply:
I think the latest change in c1fdc6 reaches the same goal as you proposed, right?

Reply:
Basically yes. The only difference is that your variant will not crash upon instantiation if neither name is provided but instead will crash later on - when the name is first accessed.

Reply:
One thing to consider: It could look odd to someone reading the code, as it's not a common pattern to only conditionally set an attribute. (Also: no docstring explaining it.) That's why I opted for the variant that makes the fallback to the class attribute explicit.

Reply:
Ah, I see, I'll change it to your version. It's also better that it fails on init, this is easier to discover in tests. I hadn't thought about it